### ISSUE TITLE
[New Primitive] `Timeline` / `Stepper`

### ISSUE THREAD

**Issue by multiwebinc** [radix-ui/primitives#2140]
### Timeline:

![image](https://github.com/radix-ui/primitives/assets/901732/62ff0ba3-1ed8-4194-a1e1-5a4ee356ced1)

### Stepper

Horizontal:

![image](https://github.com/radix-ui/primitives/assets/901732/e1d6389b-b2a5-4d82-b61d-01adc317bef8)

Vertical:

![image](https://github.com/radix-ui/primitives/assets/901732/eded8767-d05a-4cdf-b277-f9bd34e720e2)

I'm including both of these in the same feature request since visually they are pretty similar, so would likely share a lot of the code.

### Examples in other libraries

https://mui.com/material-ui/react-timeline/
https://mui.com/material-ui/react-stepper/

https://flowbite.com/docs/components/timeline/
https://flowbite.com/docs/components/stepper/

https://tailblocks.cc/ (under "Stepper") [Github](https://github.com/mertJF/tailblocks/tree/master/src/blocks/step)

https://www.skeleton.dev/components/steppers

https://chakra-ui.com/docs/components/stepper/usage

https://ant.design/components/steps
https://ant.design/docs/spec/data-display#timeline

### Who does this impact? Who is this for?

Someone who wants to show a timeline of events or add a multi-step form.

**Comment by X7md** [comment]
Any progress?

**Comment by ImanMahmoudinasab** [comment]
I'm interested in contributing to the design & development of this component.

**Comment by robinraju** [comment]
I found this on shadcn-ui:  https://github.com/shadcn-ui/ui/pull/318

**Comment by Babailan** [comment]
If only they will published this on the later releases I would be glad.

**Comment by yashr42** [comment]
any updates about this feature?

**Comment by HKA-s** [comment]
Up

**Comment by adamperlis** [comment]
Up

**Comment by majoess** [comment]
up

**Comment by adamperlis** [comment]
Also would be great if this was a feed like this: https://tailwindui.com/components/application-ui/lists/feeds

**Comment by mrvinrsk** [comment]
Up

**Comment by SukkaW** [comment]
@chaance What happened with this proposal? This is a highly required component, TBH.

**Comment by PavelMelnik94** [comment]
up

**Comment by Sachin0075** [comment]
yeah i belive this should have been in radix ui. we are building our own custom ui library and choosed radix ui as base but it seems radix dosent have components like MUI or other libraries🥲

**Comment by chaance** [comment]
This is not on the roadmap at this time. New ideas are welcome in discussions, but we need to keep issues focused on bug reports.
